### PR TITLE
Fix up GCP docs

### PR DESCRIPTION
**Pull request by zmb3**
The most notable issue with our GCP docs is that they don't tell you which specific permissions are required. This has caused problems for customers who weren't aware that we require permission to delete objects from GCS.

- Fix an incorrect link
- Be more specific about what roles/permissions are required
- Remove some text and an image that didn't meaningfully contribute

### CODE REVIEW

File: docs/pages/deploy-a-cluster/deployments/gcp.mdx [hunk 1]
```diff
@@ -16,36 +16,20 @@ Teleport Cloud manages for you.
 
 We have split this guide into:
 
-- [Teleport on GCP FAQ](#teleport-on-gcp-faq)
 - [GCP Teleport Introduction](#gcp-teleport-introduction)
 - [GCP Quickstart](#gcp-quickstart)
 
 (!docs/pages/includes/cloud/call-to-action.mdx!)
 
-## Teleport on GCP FAQ
```
Comment (zmb3) [inline docs/pages/deploy-a-cluster/deployments/gcp.mdx:25]: I didn't find this section compelling enough to keep. If someone's reading our GCP docs we probably don't need to sell them on Teleport.

File: docs/pages/deploy-a-cluster/deployments/gcp.mdx [hunk 2]
```diff
@@ -16,36 +16,20 @@ Teleport Cloud manages for you.
...
 
 This guide will cover how to setup, configure and run Teleport on GCP.
 
-GCP Services required to run Teleport in High Availability:
+The following GCP Services are required to run Teleport in high availability mode:
 
 - [Compute Engine: VM Instances with Instance Groups](#compute-engine-vm-instances-with-instance-groups)
 - [Compute Engine: Health Checks](#compute-engine-health-checks)
-- [Storage: Cloud Firestore](#storage-google-cloud-storage)
+- [Storage: Cloud Firestore](#storage-cloud-firestore)
```
Comment (zmb3) [inline docs/pages/deploy-a-cluster/deployments/gcp.mdx:32]: This has always been incorrect.

File: docs/pages/deploy-a-cluster/deployments/gcp.mdx [hunk 3]
```diff
@@ -80,34 +68,36 @@ see  [Admin Guide: Troubleshooting](../../management/admin/troubleshooting.mdx)
...
 
 ```yaml
 teleport:
   storage:
     type: firestore
     collection_name: cluster-data
-    project_id: EXAMPLE_gcp-proj-with-firestore-enabled
-    credentials_path: /var/lib/teleport/firestore_creds.json
-    audit_events_uri: ['firestore://events_table_name?projectID=example_Teleport-Project-Name&credentialsPath=/var/lib/teleport/gcs_creds.json']
+    project_id: <Var name="GCP-Project-ID" description="Your GCP project ID"/>
```
Comment (zmb3) [inline docs/pages/deploy-a-cluster/deployments/gcp.mdx:83]: Note: waiting on docs team to add a `default` attribute here.

cc @ptgott

File: docs/pages/deploy-a-cluster/helm-deployments/gcp.mdx [hunk 4]
```diff
@@ -149,8 +149,7 @@ Go to the "Service Accounts" section of Google Cloud IAM & Admin.
 | storage-bucket-creator-role | Role you just created allowing creation of storage buckets |
 | dns-updater-role | Role you just created allowing updates to Cloud DNS records |
 | Cloud Datastore Owner | Grants permissions to create Cloud Datastore collections |
-| Storage Object Creator | Allows writing of Google Cloud storage objects |
-| Storage Object Viewer | Allows reading of Google Cloud storage objects |
+| Storage Object Admin | Allows read/write/delete of Google Cloud storage objects |
```
Comment (zmb3) [inline docs/pages/deploy-a-cluster/helm-deployments/gcp.mdx:152]: The helm guide could probably use a fresh set of eyes as well - but it requires some extra permissions so for now I've opted to fix the most important error (lack of delete permissions).

File: docs/pages/deploy-a-cluster/deployments/gcp.mdx [hunk 5]
```diff
@@ -57,18 +41,22 @@ Other things needed:
...
 
 - Management Tools: Cloud Deployment Manager
-- Stackdriver Logging
+- Logging: Stackdriver
 
-We recommend setting up Teleport in High Availability mode. In High Availability mode Firestore
-stores the state of the system and Google Cloud Storage stores the audit logs.
+We recommend setting up Teleport in high availability mode.
+In high availability mode Firestore is used for cluster state
+and Google Cloud Storage is used for audit logs.
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/deployments/gcp.mdx:48]: AFAIK Firestore is used for both cluster state and audit logs - Google Cloud Storage is used for session recordings (i.e. this has always been wrong)

Comment (zmb3) [reply docs/pages/deploy-a-cluster/deployments/gcp.mdx:48]: Yep, good catch. Thanks G!

File: docs/pages/deploy-a-cluster/deployments/gcp.mdx (hunk 3 above)
Comment (zmb3) [reply docs/pages/deploy-a-cluster/deployments/gcp.mdx:83]: Ok, I'll remove the use of Var throughout this page and just use generic placeholders like we had before.

Comment (zmb3) [reply docs/pages/deploy-a-cluster/deployments/gcp.mdx]: All set. Thanks Paul!

File: docs/pages/deploy-a-cluster/deployments/gcp.mdx [hunk 6]
```diff
@@ -145,7 +225,8 @@ Follow install instructions from our [installation page](../../installation.mdx#
 We recommend configuring Teleport as per the below steps:
 
 **1. Configure Teleport Auth Server** using the below example `teleport.yaml`, and start it
-using [systemd](https://raw.githubusercontent.com/gravitational/teleport/master/examples/systemd/teleport.service) or use DEB/.RPM packages available from our [Downloads Page](https://goteleport.com/download/).
+using [systemd](https://raw.githubusercontent.com/gravitational/teleport/master/examples/systemd/teleport.service)
+or use DEB/.RPM packages available from our [Downloads Page](https://goteleport.com/download/).
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/deployments/gcp.mdx:229]:
```suggestion
or use DEB/RPM packages available from our [Downloads Page](https://goteleport.com/download/).
```

**Review by webvictim — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Review by alexfornuto — APPROVED** [review]

**Pull request merged**